commands/pull_request.go

```diff
@@ -467,6 +472,33 @@ func parsePullRequestIssueNumber(url string) string {
 	return ""
 }
 
+func prepareMessageFromCommit(commit string) (string, error) {
+	message, err := git.Show(commit)
+	if err != nil {
+		return "", err
+	}
+	message = strings.Replace(string(message), "\r\n", "\n", -1)
+
+	message = appendPullRequestTemplate(message)
```

Comment:
I just realized that we probably don't want to be using a pull request template here:

1. If `--edit` wasn't passed (off by default), the person would not even have the chance to edit the template. Submitting the contents of a non-edited template is most likely wrong.
2. If `--edit` was passed, I still don't think we want to apply the template, because the subject and the body of the given commit acts as a "template" already.

Could you amend the behavior and tests please if this makes sense to you?

Reply:
Well, one of the motivation behind the implementation is (as it's probably most often the case) to scratch my pain point. Currently, most of the time when I'm opening PR I need:

* to use HEAD or some commit in branch as "base" for PR title and body (as I'm used to write "meaningful" commits so I can reuse them for this purpose - as I want reasonable starting point when describing the pull request)
* add information which we have in PR template (i.e. I want each PR to follow the structure as defined in the template)

I'm doing it right now by running something like `my_custom_script > file && hub pull-request -F file --edit`, where `my_custom_script`:

* takes HEAD commit and add it at the beginning of file
* parse and update PR template from repo

and I just update details if something is missing/redundant in manual edit. 

So I was thinking at the beginning about adding `-C` to `hub pull-request -F` but as I've found  that the behavior is bit inconsistent, it looks to me easier to introduce it separately and eventually in another PR unify the behavior across different flags. 

One possible solution I see here - which would keep around adding PR template to PR message - might be introducing another flag like `[no]template` so even if the user does not pass `--edit`, she might decide to skip the template. I am not sure what is preferable default behavior so let me know what you would prefer. 

Anyway, in case that this solution would not work for you, I will finish the PR as you propose but it would be nice to think about a way how it would be possible to implement the features which would allow to use `hub` in use case described above. I'm definitely open to implementing them in another PRs :)

Reply:
> introducing another flag like `[no]template` so even if the user does not pass `--edit`, she might decide to skip the template. I am not sure what is preferable default behavior so let me know what you would prefer.

Thanks for explaining your use-case. I'm wary of adding more flags and configuration options such as `--[no]template`. From my standpoint, when someone's workflow becomes too specific to be sufficiently covered by hub core behavior, they should resort to using scripts to encapsulate their workflow and use `hub pull-request -F <file>` to pass contents in programmatically. You have already gone down that route, it seems.

It's also nice that some of these features would be available in core without the necessity of scripting them outside of hub, but I'm not sure what the right behavior should be here. Maybe we can add the template when `--edit` was present but skip it without. (I still think that posting a template without editing is never desireable.) Maybe I need to think about this for a while.

Reply:
> I'm wary of adding more flags and configuration options such as --[no]template

completely understand that, it's always a pain reading help/man with never ending list of flags

Agree with you on both points (specific behavior -> use custom script vs feature in core), I actually cannot say what is the right way here; it seems to me that somehow using commit message when creating PR is reasonable (as github is doing the same when you open PR using frontend).

Regarding edit/noedit behavior, personally I cannot see when/why you want to run `hub` without `--edit` (yes, you can always edit it once PR is created but) - if nothing else I would just skim through the title/body to check if it makes sense - whatever hub workflow you use. But that's obviously just my personal POV.

 > Maybe we can add the template when --edit was present but skip it without.

This would definitely works for me, but I somehow feel like if we are not making things "too complicated" / difficult to understand from user perspective (unless she reads the actual implementation).

> Maybe I need to think about this for a while.

Sure, np, let me know once you have some idea how to continue with it - I will finish PR then :)

Thanks for all the insights.